storage/innobase/trx/trx0rseg.cc

```diff
@@ -584,7 +584,11 @@ dberr_t trx_rseg_array_init()
 #endif
 	mtr_t mtr;
 	dberr_t err = DB_SUCCESS;
-
+	/* Acquiring purge queue mutex here should be fine from the deadlock
+	prevention point of view, because executing that function is a
+	prerequisite for starting the purge subsystem or any transactions. */
+	if (srv_operation != SRV_OPERATION_RESTORE)
+		purge_sys.lock_purge_queue_mutex();
```

Comment:
The comment should explain why we are avoiding the mutex acquisition in `mariadb-backup --prepare`. Maybe we should simply call `purge_sys.create()` somewhere in `xtrabackup.cc`?